docs/architecture/adr-066-rosetta-plugins.md

```diff
@@ -0,0 +1,121 @@
+# ADR-066: Rosetta Plugins (multi zone=
+
+## Changelog
+
+* May 30, 2023: Initial Draft: {changelog}
+
+## Status
+
+{DRAFT | PROPOSED} Not Implemented
+
+## Abstract
+
+Enabling Rosetta in multiple zones requires adding interfaces and codec types.
+Forking the project and implementing it would require a lot of effort. The proposed approach
+aims to load such interfaces and types in a simple and easy way instead of implementing
+Roseta as a whole
+
+## Context
+
+The Cosmos ecosystem contains multiple zones with different message types and interfaces.
+Rosetta is a tool that allows chains to be listed by Coinbase while also providing a standard
+access chain information.
+
+In order for Rosetta to work in different zones, it requires it to register the types and
+interfaces specific to the zones.
+
+## Alternatives
+
+There are a few alternatives:
+- **Implementation**: It requires the zone developers to come up with their own implementation of
+Rosetta, from top to bottom
+- **Fork**: It takes the Cosmos SDK implementation but modifies it to register the different interfaces.
+and keeping it updated by the zone developers.
+- **Reflection**: Cosmos-sdk v0.47.x provides a mechanism to reflect over interfaces and types.
+through RPc endpoints (see the HUDL tool). It requires zone owners to implement a service on
+their nodes in order to enable this feature to be used.
+- **Plugin**: Each zone just implements the configurations required (like prefixes) and loads the
+required interfaces.
+
+## Decision
+In order for Rosetta to access different zones and decode the chain data, it must have
+registered chain-specific interfaces; this involves importing such interfaces and types.
+In most cases, these types depend on the same libraries as Rosetta does, but with
+different version.
+
+Onboarding a new chain involves spending a lot of time tuning up the conflicting packages.
+Versioning (also known as dependency hell), in some cases, also means hitting a wall.
+making it impossible to onboard a new chain or, in the long term, adding more complexity as the
+amount of supported chains grows
+
+The only thing missing on the Cosmos-SDK Rosetta implementation is the ability to work with other chains.
```

Comment:
Err, I don't understand the sentence from line 51 to 54, there seem to be bits of sentence in the middle ("is just missing interfaces" and "main dependencies.")